envi/memcanvas/__init__.py

```diff
@@ -383,6 +383,40 @@ def __str__(self):
         return self.strval
 
 
+class FileBackedMemoryCanvas(StringMemoryCanvas):
+    '''
+    FileBackedMemoryCanvas writes output to a file of your choosing.  It writes
+    out the data to the file after `rollwindow` number of calls to `addText()`.
+
+    NOTE: when done writing to the canvas, be certain to call `flush()` or you
+            will lose data.
+    '''
+    def __init__(self, filename, mem, syms=None, append=True, rollwindow=100):
+        StringMemoryCanvas.__init__(self, mem, syms)
+        if append:
+            self.fd = open(filename, 'ab')
+        else:
+            self.fd = open(filename, 'wb')
+
+        self.rollwindow = rollwindow
+        self.linecount = 0
+
+    def addText(self, text, tag=None):
+        StringMemoryCanvas.addText(self, text, tag)
```

Comment:
i'll see if it works this time, just don't want to have another #613

Reply:
isn't that worse performance-wise?  we know what we're trying to call, why not access it directly rather than calling super()?  or is the magic of super() such that it's faster than just going direct?
i mean, this isn't an emulator i_* function, so it doesn't have to be *that* performant, but i'm wondering.

Reply:
So #613 happened because that's just a straight syntax error. It's not the fault of super that the parens were forgotten.

So the super isn't for this class, it's for anybody descended from this class so that method resolution happens the way that python says to and things work as expected, especially if we ever end up doing any kinda of multiple inheritance stuff.  Which I've had to track down in vivisect before, so I would like things to be easier to track down.

And in terms of performance they're identical, so there's no performance problems.

Reply:
sorry, changing `StringMemoryCanvas` to `super()` breaks it.  can you make a suggested edit that works?  

my current testing:

```
$ vdbbin -c /bin/chown
```
and from the VDB command line:

```
saveout foo1.bar stepi -V -C 1000
```

using `super()`, i get an empty file
as is, it dumps data as desired.

Reply:
For posterity:

`super().addText(text, tag)`